packages/next/shared/lib/router/router.ts

```diff
@@ -1897,10 +1897,12 @@ export default class Router implements BaseRouter {
       return { type: 'next' }
     }
 
+    const preflightHref = addLocale(options.as, options.locale)
+
     let preflight: PreflightData | undefined
     try {
       preflight = await this._getPreflightData({
-        preflightHref: options.as,
+        preflightHref,
```

Comment:
This PR causes an infinite reload for us.
We didn't have `/uk/page-2-path` configured in akamai so when we loaded `/page-1-path` (`uk` as our default locale), it tried to prefetch `/uk/page-2-path`. That failed and it reloaded/redirected the browser to `/page-1-path` and then same thing... infinite reload :|
It's a pre-flight/preload another page so if that fails, it should not reload and let the app works normally.